File: node-rest-api/todos/update.js
'use strict';

const AWS = require('aws-sdk');

const dynamoDb = new AWS.DynamoDB.DocumentClient();

module.exports.update = (event, context, callback) => {
    const timestamp = new Date().getTime();
    const data = JSON.parse(event.body);


    if (typeof data.text !== 'string' || typeof data.checked !== 'boolean') {
        console.error('ERROR en la validacion');

        callback(null, {
            statusCode: 400,
            headers: {'Content-Type': 'text/plain'},
            body: 'no se pudo actualizar el elemento',
        });

        return;
    }

    const params = {
        TableName: process.env.DYNAMODB_TABLE,
        Key: {
            id: event.pathParameters.id
        },
        ExpressionAttributeNames:{
            '#todo_text': 'text'
        },
        ExpressionAttributeValues:{
            ':text': data.text,
            ':checked': data.checked,
            ':updatedAt': timestamp
        },
        UpdateExpression: 'SET #todo_text = :text, checked = :checked, updateAt = :updateAt',
        ReturnValues: 'ALL_NEW'
    };

    dynamoDb.update(params, (error, result)=>{
        if(error){
            console.error(error);
            callback(null,{
                statusCode: error.statusCode || 501,
                headers: {'Content-Type': 'text/plain'},
                body: 'no se pudo actualizar el elemento'
            });
            return;
        }
        const response = {
            statusCode: 200,
            body: JSON.stringify(result.Attributes),
        };
        callback(null, response);
    });
};

File: api-paython/update.py
import json
import os
import time
import logging
import decimalencoder
import boto3

dynamoDb = boto3.resource('dynamodb')


def update(event, context):
    data = json.loads(event['body'])

    if 'nombre' not in data or 'checked' not in data or 'apellidos' not in data or 'edad' not in data or 'email' not in data:
        logging.error('Validacion Fallida')
        raise Exception("No se pudo actualizar el elemento")
        return

    timestamp = int(time.time() * 1000)

    table = dynamoDb.Table(os.environ['python_users'])

    result = table.update_item(
        Key={
            'id': event['pathParameters']['id']
        },
        ExpressionAttributeNames={
            '#todo_nombre': 'nombre',
            '#todo_apellidos': 'apellidos',
            '#todo_edad': 'edad',
            '#todo_email': 'email'
        },
        ExpressionAttributeValues={
            ':nombre': data['nombre'],
            ':apellidos': data['apellidos'],
            ':edad': data['edad'],
            ':email': data['email'],
            ':checked': data['cheked'],
            ':actualizado': timestamp
        },
        UpdateExpression='SET #todo_nombre = :nombre,'
        'apellidos = :apellidos,'
        'edad = :edad,'
        'email = :email,'
        'checked = :checked,'
        'actualizado = :actualizado',
        ReturnValues='ALL_NEW',
    )

    response = {
        "statusCode": 200,
        "body": json.dumps(result['Attributes'],
                           cls=decimalencoder.DecimalEncoder)
    }

    return response
